lib/src/components/NxTransferListHalf/__tests__/NxTransferListHalf.test.tsx

```diff
@@ -4,496 +4,507 @@
+      await user.click(button);
 
-      expect(onMoveAll).toHaveBeenCalledWith([1, 2, 4, 5]);
+      expect(onMoveAll).toHaveBeenCalledWith(['1', '2', '4', '5']);
     });
   });
 
   describe('reordering', function() {
-    it('renders move up and down buttons inside .nx-transfer-list__item in the correct state', function() {
-      const items = [{
-            id: 1,
-            displayName: 'top'
+    it('renders move up and down buttons inside .nx-transfer-list__item when allowReordering is true',
+        async function() {
+          const dataItems = [{
+            id: '1',
+            displayName: 'foo'
           }, {
-            id: 3,
-            displayName: 'middle'
+            id: '2',
+            displayName: 'bar'
           }, {
-            id: 4,
-            displayName: 'bottom'
-          }],
-          component = getMounted({ allowReordering: true, items }),
-          list = component.find('.nx-transfer-list__item-list');
-
-      const firstItem = list.find('.nx-transfer-list__item').at(0),
-          secondItem = list.find('.nx-transfer-list__item').at(1),
-          thirdItem = list.find('.nx-transfer-list__item').at(2),
-          firstItemMoveUpButton = firstItem.find('.nx-btn').at(0),
-          firstItemMoveDownButton = firstItem.find('.nx-btn').at(1),
-          secondItemMoveUpButton = secondItem.find('.nx-btn').at(0),
-          secondItemMoveDownButton = secondItem.find('.nx-btn').at(1),
-          thirdItemMoveUpButton = thirdItem.find('.nx-btn').at(0),
-          thirdItemMoveDownButton = thirdItem.find('.nx-btn').at(1);
-
-      expect(firstItem).toExist();
-      expect(firstItemMoveUpButton).toExist();
-      expect(firstItemMoveDownButton).toExist();
-      expect(firstItemMoveUpButton).toHaveClassName('disabled');
-      expect(firstItemMoveDownButton).not.toHaveClassName('disabled');
-
-      expect(secondItem).toExist();
-      expect(secondItemMoveUpButton).toExist();
-      expect(secondItemMoveDownButton).toExist();
-      expect(secondItemMoveUpButton).not.toHaveClassName('disabled');
-      expect(secondItemMoveDownButton).not.toHaveClassName('disabled');
-
-      expect(thirdItem).toExist();
-      expect(thirdItemMoveUpButton).toExist();
-      expect(thirdItemMoveDownButton).toExist();
-      expect(thirdItemMoveUpButton).not.toHaveClassName('disabled');
-      expect(thirdItemMoveDownButton).toHaveClassName('disabled');
+            id: '3',
+            displayName: 'foobar'
+          }];
+
+          const { container, rerender } = quickRender({ allowReordering: true, items: dataItems}),
+              middleItem = container.querySelectorAll<HTMLElement>('.nx-transfer-list__item')[1],
+              buttons = within(middleItem).getAllByRole('button', { hidden: true });
+
+          // assert both a move up and move down button are present
+          expect(buttons.length).toBe(2);
+          await waitFor(() => expect(buttons[0]).toHaveAccessibleName('Move Up'));
```

Comment:
Does this actually work now, without a call to `runTimers` before it? What about without the `waitFor`s?

Reply:
You're right, I needed to replace that. Updated here and working: [ef7faff](https://github.com/sonatype/sonatype-react-shared-components/pull/880/commits/ef7faffad6cfd011c7ca3107d0dd3a235050ff34)